src/snowflake/snowpark/dataframe_writer.py

```diff
@@ -90,6 +91,11 @@ def save_as_table(
 
                 "ignore": Ignore this operation if data already exists.
 
+            column_order: When ``mode`` is "append", data will be inserted into the target table by matching column sequence or column name. Default is "index".
+
+                "index": Data will be inserted into the target table by column sequence.
+                "name": Data will be inserted into the target table by matching column names. If the target table has more columns than the source DataFrame, use this one.
```

Comment:
should we consider str Enum for this mode?

Reply:
This is a good suggestion. We haven't used Enum. We may add Enum in the future to APIs where it's good. But don't want to single this one out for now.